ALEC-94: Update health check command (#130)
OpenNMS/alec: docs/modules/admin/pages/distributed_install.adoc

== OpenNMS configuration

Follow these steps to configure your OpenNMS instance for a distributed installation of ALEC.

> The steps should be completed using the OpenNMS Karaf shell.

Reviewer: ```suggestion
You should use the OpenNMS Karaf shell to complete these steps. 
```

NOTE: This procedure assumes that you are already using the Kafka strategy for the Sink API (see <<prerequisites, Prerequisites>>).

. Configure the Kafka Producer to point to your Kafka broker:
+
[source, shell]
echo 'bootstrap.servers=127.0.0.1:9092' > "$OPENNMS_HOME/etc/org.opennms.features.kafka.producer.client.cfg"

. Disable incremental alarm suppression:
+
[source, shell]
echo 'suppressIncrementalAlarms=false' >> "$OPENNMS_HOME/etc/org.opennms.features.kafka.producer.cfg"

. Edit `$OPENNMS_HOME/etc/org.apache.karaf.features.cfg` and add `opennms-kafka-producer` to the `featuresBoost` property.
. Enable consumption of Kafka events:
+
[source, shell]
echo 'org.opennms.netmgt.eventd.sink.enable=true' > "$OPENNMS_HOME/etc/opennms.properties.d/event-sink.properties"

== Install ALEC in OpenNMS

Follow these steps to install ALEC on your OpenNMS instance.
Make sure that you install the package repository on all systems that are running OpenNMS.

. Install the plugin package on your OpenNMS server:
+
[source, shell]
yum -y install opennms-alec-plugin

> . Configure the feature set to be installed when OpenNMS starts:

Reviewer: ```suggestion
. Configure the feature set to install when OpenNMS starts:
```

+
[source, shell]
echo 'alec-opennms-distributed wait-for-kar=opennms-alec-plugin' > "$OPENNMS_HOME/etc/featuresBoot.d/alec.boot"

== Install ALEC in Sentinel

Follow these steps to install ALEC on your Sentinel instance.
Make sure that you install the package repository on all systems that are running Sentinel.

. Install the plugin package on your Sentinel systems:
+
[source, shell]
yum -y install sentinel-alec-plugin

. Configure Kafka Stream (the Kafka consumer):
+
[source, shell]
----
echo 'bootstrap.servers=127.0.0.1:9092
commit.interval.ms=5000' > "$SENTINEL_HOME/etc/org.opennms.alec.datasource.opennms.kafka.streams.cfg"
----

. Configure the Kafka producer:
+
[source, shell]
----
echo 'bootstrap.servers=127.0.0.1:9092' > "$SENTINEL_HOME/etc/org.opennms.alec.datasource.opennms.kafka.producer.cfg"
----
+
TIP: Kafka Stream reads alarms and inventory, while the Kafka producer sends events.

. Configure ZooKeeper for distributed coordination:
+
[source, shell]
echo 'connectString=127.0.0.1:2181' > "$SENTINEL_HOME/etc/org.opennms.features.distributed.coordination.zookeeper.cfg"

> . Configure the feature set to be installed when Sentinel starts:

Reviewer: ```suggestion
. Configure the feature set to install when Sentinel starts:
```
